iop/cdvd/cdfs/src/cdfs_iop.c

```diff
@@ -0,0 +1,1263 @@
+            if (tocEntryPointer->length == 0) {
+#ifdef DEBUG
+                printf("Got a null pointer entry, so either reached end of dir, or end of sector\n");
+                printf("Offset into cache = %d bytes\n", (char *)tocEntryPointer - cacheInfoDir.cache);
+#endif
+
+                tocEntryPointer = (struct DirTocEntry *)(cacheInfoDir.cache + (((((char *)tocEntryPointer - cacheInfoDir.cache) / 2048) + 1) * 2048));
+            }
+
+            if (tocEntryPointer >= (struct DirTocEntry *)(cacheInfoDir.cache + (cacheInfoDir.cache_size * 2048))) {
+                // reached the end of the cache block
+                break;
+            }
+
+            copyToTocEntry(tocEntry, tocEntryPointer);
+
+            if (strcasecmp(tocEntry->filename, filename) == 0) {
+#ifdef DEBUG
+                printf("Found a matching file\n\n");
+#endif
+                // and it matches !!
+                return TRUE;
+            }
+#ifdef DEBUG
+            printf("Non-matching file - looking for %s , found %s\n", filename, tocEntry->filename);
+#endif
+        }  // end of for loop
+
+#ifdef DEBUG
+        printf("Reached end of cache block\n");
+#endif
+        // cache the next block
+        cdfs_cacheDir(pathname, CACHE_NEXT);
+    }
+
+// we've run out of dir blocks to cache, and still not found it, so fail
+
+#ifdef DEBUG
+    printf("cdfs_findfile: could not find file\n");
+#endif
+
+    return FALSE;
+}
+
+/********************
+* Optimised CD Read *
+********************/
+
+int cdfs_readSect(u32 lsn, u32 sectors, void *buf) {
+    int retry;
+    int result = 0;
+    cdReadMode.trycount = 32;
+
+    for (retry = 0; retry < 32; retry++)  // 32 retries
+    {
+        if (retry <= 8)
+            cdReadMode.spindlctrl = 1;  // Try fast reads for first 8 tries
+        else
+            cdReadMode.spindlctrl = 0;  // Then try slow reads
+
+        if (!isValidDisc())
+            return FALSE;
+
+        sceCdDiskReady(0);
+
+        if (!sceCdRead(lsn, sectors, buf, &cdReadMode)) {
+            // Failed to read
+            if (retry == 31) {
+                // Still failed after last retry
+                memset(buf, 0, (sectors << 11));
+                // printf("Couldn't Read from file for some reason\n");
+                return FALSE;  // error
+            }
+        } else {
+            // Read okay
+            sceCdSync(0);
+            break;
+        }
+
+        result = sceCdGetError();
+        if (result == 0)
+            break;
+    }
+
+    cdReadMode.trycount = 32;
+    cdReadMode.spindlctrl = 1;
+
+    if (result == 0)
+        return TRUE;
+
+    memset(buf, 0, (sectors << 11));
+
+    return FALSE;  // error
+}
+
+// This is the getdir for use by the EE RPC client
+// It DMA's entries to the specified buffer in EE memory
+int cdfs_getDir(const char *pathname, const char *extensions, enum CDFS_getMode getMode, struct TocEntry tocEntry[], unsigned int req_entries) {
```

Comment:
It looks like this function transfers to the EE memory, but it's used from IOP. Is this correct?

Reply:
Yeah, you were right, it was using EE memory. Now I have updated the PR and remove this logic